### DISCUSSION TITLE
Scene-viewer not showing changes on model in AR view

### DISCUSSION THREAD

**Discussion by danilokrulj** [google/model-viewer#3924] [Q&A]
I have made a simple configurator where you can chose between different textures of the model, when you decide which texture you like and you proceed to view the model in AR the changes don't take place. The original model gets instanced instead of the model with the new texture. I remember this used to work before, when I switch to webxr the changes are visible in AR. Did something happen with scene-viewer? Changes are also visible on iOS quick-look.

**Comment by elalish** [comment]
This has never worked in Scene Viewer because it re-downloads the model from your original URL. Is there any reason you don't want to use WebXR mode? It has quite a few advantages.

**Reply by gjhatzis** [reply]
Hello @danilokrulj  and @elalish, I face the same issue with scene-viewer.
I prefer scene viewer too, since it tracks 3d models and mobile movement more accurately on real space, compared to WebXR.
What's your opinion on that?

**Reply by danilokrulj** [reply]
As usual elalish is right. SceneViewer didnt support this behaviour I just havent worked with MV for some time and forgot what the limitations are. As for the more accurate tracking in SceneViewer I must admit I havent noticed that feature. @elalish Is it maybe possible to save the modified model as a blob or something else and than serve it as a new "src" value?

**Reply by elalish** [reply]
Unfortunately you can't pass a blob to SceneViewer like you can to QuickLook. WebXR and SceneViewer both use the same ARCore code for tracking, so they should be the same. That said, they may use ARCore slightly differently and if WebXR is worse, that's likely a bug. If you can file something with screen recordings comparing the two in the same environment, that would be very helpful.
